### DISCUSSION TITLE
How theming - css variables works?

### DISCUSSION THREAD

**Discussion by yeion7** [adobe/react-spectrum#1945] [Q&A]
I have been trying to wrap my head around how theming works, but there is something that I can't get it

For example [tooltip](https://react-spectrum.adobe.com/react-spectrum/Tooltip.html)

The css module that is importing the component definition is using these [css vars](
https://github.com/adobe/react-spectrum/blob/main/packages/%40adobe/spectrum-css-temp/components/tooltip/index.css#L35-L41) 

![image](https://user-images.githubusercontent.com/9127504/119345763-f5b29000-bc5e-11eb-8cfd-01ce0054497d.png)

But then, in the css default theme definition [these vars](https://github.com/adobe/react-spectrum/blob/main/packages/%40adobe/spectrum-css-temp/vars/spectrum-large.css#L397-L407)

![image](https://user-images.githubusercontent.com/9127504/119345910-1da1f380-bc5f-11eb-9712-0cf21d92c0a8.png)

The more confusing thing for me is

Storybook seems to be having the same variables than the css module definition

![image](https://user-images.githubusercontent.com/9127504/119346094-635ebc00-bc5f-11eb-8d76-7a95925dd956.png)

But when I installed `@adobe/react-spectrum@3.10.0` and I used the `Tooltip` component, it's not using those variables

![image](https://user-images.githubusercontent.com/9127504/119346302-a7ea5780-bc5f-11eb-8a34-ec247ec67921.png)

([codesandbox](https://codesandbox.io/s/gracious-hypatia-yh66p?file=/src/App.js) to inspect tooltip css)

How theming works? there is a way to how what variables includes each component?

**Comment by snowystinger** [comment]
Ah, those variables are coming from `node_modules/@spectrum-css/vars/dist/components/spectrum-tooltip.css` when we build
We haven't begun keeping the tokens as API yet because we're still working with our CSS team to solve some issues
For now, be careful using them
